modules/internal/ChapelTuple.chpl

```diff
@@ -219,7 +219,7 @@ module ChapelTuple {
     if CHPL_WARN_TUPLE_ITERATION == "true" then
       compilerWarning("Iterating over tuples. If you intended to use zippered iteration, add the new keyword 'zip' before the tuple of iteratable expressions.");
 
-    foreach i in 0..#this.size {
+    foreach i in 0..#this.size with (ref this) {
```

Comment:
While giving me a quick walkthrough of the diff earlier, I mentioned that this looked strange. I still do, but I think this is consistent with what happens with `forall`s. The way we treat `this` is a bit cumbersome to me, but we do require this explicitly here.

Related: https://github.com/chapel-lang/chapel/issues/23627

Reply:
Not sure exactly what you're finding strange about it.

Given that this is inside of an iterator that yields `ref`s it would make sense that we'd need `this(i)` (or maybe `this.data` or something) to have `ref` intent but AFAIK there's no way to specify that. The current behavior (even for `forall`) is that you can't give intents on fields of objects (like say `this.prop`) but you can on the object itself as a whole (like `this`).